cro-clib/src/transaction_build.rs

```diff
@@ -0,0 +1,294 @@
+use crate::types::get_string;
+use crate::types::{CroAddress, CroAddressPtr, CroResult};
+use crate::types::{CroTx, CroTxPtr};
+use chain_core::init::coin::Coin;
+pub use chain_core::init::network::Network;
+use chain_core::tx::data::access::{TxAccess, TxAccessPolicy};
+use chain_core::tx::data::address::ExtendedAddr;
+use chain_core::tx::data::attribute::TxAttributes;
+use chain_core::tx::data::input::TxoPointer;
+use chain_core::tx::data::output::TxOut;
+use chain_core::tx::data::Tx;
+use chain_core::tx::data::TxId;
+use chain_core::tx::witness::TxInWitness;
+use chain_core::tx::TransactionId;
+use client_common::MultiSigAddress;
+use client_common::{ErrorKind, Result, ResultExt};
+use client_common::{PrivateKey, PublicKey};
+use parity_scale_codec::Encode;
+
+use std::collections::BTreeSet;
+use std::os::raw::c_char;
+use std::ptr;
+use std::str::FromStr;
+/// create tx
+#[no_mangle]
+/// # Safety
+pub unsafe extern "C" fn cro_create_tx(tx_out: *mut CroTxPtr) -> CroResult {
+    let tx = CroTx::default();
+    let tx_box = Box::new(tx);
+    ptr::write(tx_out, Box::into_raw(tx_box));
+    CroResult::success()
+}
+
+fn do_cro_tx_add_txin(
+    tx: &mut CroTx,
+    txid_hex: &str,
+    txindex: u16,
+    addr: &str,
+    coin: u64,
+) -> Result<()> {
+    let txid = hex::decode(&txid_hex).chain(|| {
+        (
+            ErrorKind::DeserializationError,
+            "Unable to decode hex of txid",
+        )
+    })?;
+    assert!(32 == txid.len());
+
+    let mut txid_bytes: [u8; 32] = [0; 32];
+    txid_bytes.copy_from_slice(&txid[0..32]);
+    let txin_pointer = TxoPointer::new(txid_bytes, txindex as usize);
+    let txin = TxOut::new(
+        ExtendedAddr::from_str(&addr).chain(|| {
+            (
+                ErrorKind::DeserializationError,
+                "Unable to decode extended addr",
+            )
+        })?,
+        Coin::new(coin).chain(|| (ErrorKind::DeserializationError, "Unable to decode coin"))?,
+    );
+
+    tx.txin_pointer.push(txin_pointer);
+    tx.txin_witness.push(None);
+    tx.txin.push(txin);
+    Ok(())
+}
+
+/// add txin
+/// txid_string: 64 length hex-char , 32 bytes
+/// addr_string: transfer address
+/// cro_string: cro unit , example 0.0001
```

Comment:
there's no `cro_string` anymore -- so just clarify it's the amount in carsons / base units (1 = 0.0000_0001)

Reply:
ok

Reply:
ok, i'll change it